src/material/list/selection-list.ts

```diff
@@ -373,8 +380,26 @@ export class MatSelectionList extends _MatSelectionListMixinBase implements CanD
   }
   private _disabled: boolean = false;
 
+  /** Whether selection is limited to one or multiple items (default multiple). */
+  @Input()
+  get multiple(): boolean { return this._multiple; }
+  set multiple(value: boolean) {
+    const newValue = coerceBooleanProperty(value);
+
+    if (newValue !== this._multiple) {
+      this._multiple = newValue;
+      this.selectedOptions = new SelectionModel(this._multiple, this._selectedOptions.selected);
```

Comment:
This is fragile, because any external subscribers to the old selection model won't be transferred to the new one. We could handle this in the same way as `mat-select` where we don't allow for the `multiple` value to be changed after initialization. In general it's not something that you'd be changing dynamically anyway and it would simplify the changes below.

Reply:
Also got rid of the getter/setter for selectedOptions (which itself remains fragile as a non-readonly public member)